src/editor/storage/clipboard-context-menu.ts

```diff
@@ -0,0 +1,350 @@
+            return false;
+        }
+
+        const items = editor.call('selector:items');
+        if (!items.length) return false;
+
+        clipboard.value = {
+            type: type,
+            value: items[0].get(path)
+        };
+
+        if (elementHighlighted) {
+            editor.call('clipboard:flashElement', elementHighlighted);
+        }
+
+        return true;
+    });
+
+    editor.method('clipboard:paste', (path, type) => {
+        if (!editor.call('clipboard:validPaste', path, type)) {
+            return false;
+        }
+
+        // should have at least one item in selector
+        const items = editor.call('selector:items');
+        if (!items.length) return false;
+
+        const paste = clipboard.value;
+
+        // TODO
+        // verify if value is actually valid based on type
+
+        // store list of records and their values before modifying for history undo/redo
+        const records = [];
+
+        for (let i = 0; i < items.length; i++) {
+            // create history records
+            records.push({
+                item: items[i],
+                path: path,
+                valueOld: items[i].get(path),
+                valueNew: paste.value
+            });
+
+            // paste new value
+            items[i].history.enabled = false;
+            items[i].set(path, paste.value);
+            items[i].history.enabled = true;
+
+            // TODO
+            // setting render-component asset does not update materials - bug in render component inspector
+        }
+
+        // custom undo/redo to support multi-selection
+        editor.api.globals.history.add({
+            name: 'clipboard.paste',
+            combine: false,
+            undo: () => {
+                for (let i = 0; i < records.length; i++) {
+                    const item = records[i].item.latest();
+                    if (!item) continue;
+                    item.history.enabled = false;
+                    item.set(records[i].path, records[i].valueOld);
+                    item.history.enabled = true;
+                }
+            },
+            redo: () => {
+                for (let i = 0; i < records.length; i++) {
+                    const item = records[i].item.latest();
+                    if (!item) continue;
+                    item.history.enabled = false;
+                    item.set(records[i].path, records[i].valueNew);
+                    item.history.enabled = true;
+                }
+            }
+        });
+
+        if (elementHighlighted) {
+            editor.call('clipboard:flashElement', elementHighlighted);
+        }
+
+        return true;
+    });
+
+    // flash dom element when copied/pasted
+    editor.method('clipboard:flashElement', (domElement: Element) => {
+        domElement.classList.add('pcui-highlight-flash');
+        setTimeout(() => {
+            domElement.classList.remove('pcui-highlight-flash');
+        }, 250);
+    });
```

Comment:
PCUI has a function `Element#flash` but I guess some of the UI is still legacy UI....is that why you're having to do this?

Reply:
They are different. Element#flash uses outline on element. While it is ok on many elements like button or input, it does not look good on LabelGroup element due to lack of padding. So pcui-highlight-flash class implements highlight with padding to better outline whole field.